File: src/me/Qball/Wild/Checks.java
package me.Qball.Wild;

import org.bukkit.Material;
import org.bukkit.block.Biome;
import org.bukkit.entity.Player;

public class Checks{
	public static boolean Water;
	public static boolean inNether;
	public static boolean inEnd;
	public static boolean loaded;
	 public static boolean getLiquid(int x,int z, Player target)
	  {
		  int Y = 0;
		 for (int i = 255; i > 0;i--)
		 {
			 if(!target.getWorld().getBlockAt(x, i, z).getType().equals((Material.AIR)))
			 {
				  Y = i;
				  break;
			 }
		 }
		 
		  if (target.getWorld().getBlockAt(x,Y,z).isLiquid()||target.getWorld().getBiome(x,z).equals(Biome.OCEAN)||target.getWorld().getBiome(x, z).equals(Biome.DEEP_OCEAN))
	      {
	    	  Water = true;
	      }
		  else
		  {
			  Water = false;
		  }
		  return Water;
	  }
	  public static boolean inNether(int tempx,int tempz, Player target)
	  {
		 
		  if (target.getWorld().getBiome(tempx, tempz) == Biome.HELL)
	      {
	    	  inNether = true; 
	      }
		  else
		  {
			  inNether = false;
		  }
		  return inNether;
	  }
	  public static boolean inEnd(int tempx,int tempz, Player target)
	  {
		 
		  if (target.getWorld().getBiome(tempx, tempz) == Biome.SKY)
	      {
	    	  inEnd = true;
	      }
		  else
		  {
			  inEnd = false;
		  }
		  return inEnd;
	  }
	  public static void ChunkLoaded(int tempx, int tempz, Player target)
	  {
		 
		
		if (target.getWorld().isChunkLoaded(tempx,tempz) == true)
		  {
			
		  }
		  else
		  {
			  target.getWorld().getChunkAt(tempx, tempz).load();
		  }
	  }	 
	  public static int getSoildBlock(int tempx, int tempz, Player target)
	  {
		 int Y = 0;
		  for (int y = 256; y>= 0; y --)
		  {
			 Y = y;
			 if(!target.getWorld().getBlockAt(tempx, Y, tempz).isEmpty())
			 {
				Y+=2;
				break;
			 }
		  }
		 return Y;
	  }

}

File: src/main/kotlin/me/qball/wild/util/Checks.kt
package me.qball.wild.util

import me.qball.wild.Wild
import org.bukkit.Location
import org.bukkit.block.Biome
import org.bukkit.entity.Player


object Checks {
    private val plugin = Wild.instance
    fun getLiquid(loc: Location) : Boolean {
        loc.y = loc.blockY - 2.0
        var x = loc.blockX
        var z = loc.blockZ
        return loc.world.getBlockAt(loc).isLiquid || loc.world.getBiome(x, z) == Biome.OCEAN || loc.world.getBiome(x, z) ==
                Biome.DEEP_OCEAN
    }

    fun inNether(target: Player) : Boolean {
        val x = target.location.blockX
        val z = target.location.blockZ
        return target.world.getBiome(x, z) == Biome.HELL
    }

    fun inEnd(target: Player) : Boolean {
        val x = target.location.blockX
        val z = target.location.blockZ
        return target.world.getBiome(x, z) == Biome.SKY
    }

    fun loadChunkIfNotLoaded(target: Player) {
        val x = target.location.blockX
        val z = target.location.blockZ
        if(!target.world.isChunkLoaded(x, z)) {
            target.world.loadChunk(x, z)
        }
    }

    fun getSolidBlockY(x: Int, z: Int, target: Player) : Int {

        var y = 0
        for(Y in 0..256) {
            if(!target.world.getBlockAt(x, y, z).isEmpty) {
                y = Y
                break
            }
        }
        return y
    }

    fun isInWorld(target: Player) : Boolean {
        val sec = plugin.config.getConfigurationSection("Worlds")
        return sec.contains(target.world.name)
    }

    fun blacklistBiome(loc: Location) : Boolean {
        val biomes = plugin.config.getStringList("Blacklisted_Biomes")
        if(biomes.size == 0) return false
        for(i in 0..biomes.size) {
            if(loc.block.biome == Biome.valueOf(biomes.get(i).toUpperCase())) {
                return true
            }
        }
        return false
    }
}


